test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
+    """Tests for getting a System"""
+
+    def test_list(self):
+        """Test listing Systems"""
+
+        self.mock_list()
+        self.call_list(str(ObjectId()))
+        self.check_list_success()
+
+
+class UpdateDSL(SystemServiceDSL):
+    """Base class for update tests"""
+
+    _stored_system: Optional[SystemOut]
+    _system_patch: SystemPatchSchema
+    _expected_system_in: SystemIn
+    _expected_system_out: MagicMock
+    _updated_system_id: str
+    _updated_system: MagicMock
+    _update_exception: pytest.ExceptionInfo
+
+    def mock_update(self, system_id: str, system_patch_data: dict, stored_system_post_data: Optional[dict]):
+        """Mocks repository methods appropriately to test the 'update' service method
+
+        :param system_id: ID of the system that will be obtained
+        :param system_patch_data: Dictionary containing the patch data as would be required for a
+                                  SystemPatchSchema (i.e. no id, code or created and modified times required)
+        :param stored_system_post_data: Dictionary containing the system data for the existing stored System
+                                        as would be required for a SystemPostSchema (i.e. no id, code or created and
+                                        modified times required)
+        """
+
+        # Stored system
+        self._stored_system = (
+            SystemOut(
+                **SystemIn(
+                    **stored_system_post_data, code=utils.generate_code(stored_system_post_data["name"], "system")
+                ).model_dump(),
+                id=CustomObjectId(system_id),
+            )
+            if stored_system_post_data
+            else None
+        )
+        self.test_helpers.mock_get(self.mock_system_repository, self._stored_system)
+
+        # Patch schema
+        self._system_patch = SystemPatchSchema(**system_patch_data)
+
+        # Updated system
+        self._expected_system_out = MagicMock()
+        self.test_helpers.mock_update(self.mock_system_repository, self._expected_system_out)
+
+        # Construct the expected input for the repository
+        merged_system_data = {**(stored_system_post_data or {}), **system_patch_data}
+        self._expected_system_in = SystemIn(
+            **merged_system_data,
+            code=utils.generate_code(merged_system_data["name"], "system"),
+        )
+
+    def call_update(self, system_id: str):
+        """Calls the SystemService `update` method with the appropriate data from a prior call to `mock_update`"""
+
+        self._updated_system_id = system_id
+        self._updated_system = self.system_service.update(system_id, self._system_patch)
+
+    def call_update_expecting_error(self, system_id: str, error_type: type):
+        """Calls the SystemService `update` method with the appropriate data from a prior call to `mock_update`
+        while expecting an error to be raised"""
+
+        with pytest.raises(error_type) as exc:
```

Comment:
It's complaining about the type - `Expected type 'tuple[Any, ...] | Any' (matched generic type 'Type[E] | tuple[Type[E], ...]'), got 'type' instead`.